### DISCUSSION TITLE
Is there a way to set the texture wrapping?

### DISCUSSION THREAD

**Discussion by FishHawk** [gfx-rs/wgpu#2605] [Q&A] [answered]
OpenGL can set texture warping like this:
``` c
glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT);
```
Is there something similar in wgpu?

**Comment by kpreid** [comment, marked as answer]
In OpenGL, these settings are part of the texture, but in wgpu they are part of a separately created _sampler_, which you will use together with the texture in the shader. You construct a sampler with a [`SamplerDescriptor`](https://docs.rs/wgpu/latest/wgpu/struct.SamplerDescriptor.html) whose `address_mode_*` fields are the control you're looking for.
